src/Features/CSharp/Portable/CodeFixes/AddImport/CSharpAddImportCodeFixProvider.cs

```diff
@@ -465,22 +465,24 @@ protected override string GetDescription(INamespaceOrTypeSymbol namespaceSymbol,
                     SyntaxNode newRoot = root.AddUsingDirective(
                                     fullyQualifiedUsingDirective, contextNode, placeSystemNamespaceFirst,
                                     Formatter.Annotation);
-                    var newDocument = document.WithSyntaxRoot(newRoot);
-                    var newSemanticModel = await newDocument.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
-                    newRoot = await newDocument.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
                     var newUsing = newRoot
-                        .DescendantNodes().OfType<UsingDirectiveSyntax>().Where(uds => uds.IsEquivalentTo(fullyQualifiedUsingDirective, topLevel: true)).Single();
-                    var speculationAnalyzer = new SpeculationAnalyzer(newUsing.Name, simpleUsingDirective.Name, newSemanticModel, cancellationToken);
-                    if (speculationAnalyzer.ReplacementChangesSemantics())
+                        .DescendantNodes().OfType<UsingDirectiveSyntax>()
+                        .Where(uds => uds.IsEquivalentTo(fullyQualifiedUsingDirective, topLevel: true))
+                        .Single();
+                    newRoot = newRoot.TrackNodes(newUsing);
+                    var options = document.Project.Solution.Workspace.Options;
+                    var documentWithSyntaxRoot = document.WithSyntaxRoot(newRoot);
+                    var simplifiedDocument = await Simplifier.ReduceAsync(documentWithSyntaxRoot, newUsing.Span, options, cancellationToken).ConfigureAwait(false);
+                    newRoot = await simplifiedDocument.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+                    var simplifiedUsing = newRoot.GetCurrentNode(newUsing);
```

Comment:
Are we confident that `TrackNodes` will always preserve this using through simplification?